src/mesh/mesh.hpp

```diff
@@ -168,7 +170,15 @@ class Mesh {
   std::vector<LogicalLocation> GetLocList() const noexcept { return loclist; }
 
   void OutputMeshStructure(const int dim, const bool dump_mesh_structure = true);
-
+  
+    
+  
+  // Ordering here is important to prevent deallocation of pools before boundary 
+  // communication buffers 
```

Comment:
This is a little disturbing. Can you explain the issue?

Reply:
Destructors of member objects are called in reverse order of which they are declared. `boundary_comm_map` points to arrays contained in `pool_map`, so when mesh object is destroyed `boundary_comm_map` needs to be destroyed first. We could get around this by putting `pool_map` somewhere else I guess.

Reply:
oh I see. I mean I think it's fine so long as we document it. Order mattering in variable declarations is a pretty bad code smell though.